public/app/core/components/AccessControl/PermissionList.tsx

```diff
@@ -1,19 +1,36 @@
-import React from 'react';
+import React, { useMemo } from 'react';
 
 import { PermissionListItem } from './PermissionListItem';
 import { ResourcePermission } from './types';
 
 interface Props {
   title: string;
+  compareKey: 'builtInRole' | 'userLogin' | 'team';
   items: ResourcePermission[];
   permissionLevels: string[];
   canSet: boolean;
   onRemove: (item: ResourcePermission) => void;
   onChange: (resourcePermission: ResourcePermission, permission: string) => void;
 }
 
-export const PermissionList = ({ title, items, permissionLevels, canSet, onRemove, onChange }: Props) => {
-  if (items.length === 0) {
+export const PermissionList = ({ title, items, compareKey, permissionLevels, canSet, onRemove, onChange }: Props) => {
+  const computed = useMemo(() => {
+    const keep: { [key: string]: ResourcePermission } = {};
+    for (let item of items) {
+      const key = item[compareKey]!;
+      if (!keep[key]) {
+        keep[key] = item;
+        continue;
+      }
+
+      if (item.actions.length > keep[key].actions.length) {
```

Comment:
comment: this feels a little bit cheeky. I think there could be a case in the future where user friendly permissions are not necessarily subsets of one another. But it works for now, so let's just go with it :)

Reply:
Yes I know and we should look into how rework this ui to allow more control (i.e. not use friendly permissions)